lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -663,8 +663,25 @@ sub done {
     $validation->optional('result');
     $validation->optional('reason');
     $validation->optional('newbuild')->num(1);
+    $validation->optional('worker_id')->num(0, undef);
     return $self->reply->validation_error({format => 'json'}) if $validation->has_error;
 
+    # check whether the specified worker matches the actually assigned one; refuse the update if not
+    if (my $specified_worker_id = $validation->param('worker_id')) {
+        my $assigned_worker_id = $job->assigned_worker_id;
+        if (defined $assigned_worker_id) {
+            return $self->render(
+                status => 400,
+                json => {error => "Refusing to set result because job is now assigned to worker $assigned_worker_id)."}
+            ) if $assigned_worker_id != $specified_worker_id;
+        }
+        else {
+            return $self->render(
+                status => 400,
+                json   => {error => 'Refusing to set result because job has been re-scheduled.'});
```

Comment:
What does that mean? Did the job already get re-scheduled for that worker at this point?

```suggestion
                json   => {error => 'Refusing to set result because job has already been re-scheduled.'});
```

Reply:
Yes, it has "already" been re-scheduled at this point. I thought this would be clear by using present progressive but I can add an explicit "already" of course.